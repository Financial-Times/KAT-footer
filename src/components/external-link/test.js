import React from 'react';
import { mount, shallow } from 'enzyme';
import ExternalLink from './index.js';

describe('ExternalLink', () => {
  const testData = {
    label: 'test test',
    href: 'test test'
  };

  test('matches snapshot', () => {
    const wrapper = shallow(<ExternalLink label={testData.label} href={testData.href}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('component', () => {
    const wrapper = mount(<ExternalLink label={testData.label} href={testData.href}/>);

    const content = <h6 className="o-footer__external-link o-footer__matrix-title">
      <a className="o-footer__more-from-ft o-footer__matrix-title" href={testData.href}>{testData.label}</a>
    </h6>;

    test('has the right content', () => {
      expect(wrapper.contains(content)).toEqual(true);
    });
  });
});
