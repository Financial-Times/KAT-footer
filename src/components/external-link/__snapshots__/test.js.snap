// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink matches snapshot 1`] = `
"<h6 className=\\"o-footer__external-link o-footer__matrix-title\\">
<a className=\\"o-footer__more-from-ft o-footer__matrix-title\\" href=\\"test test\\">
test test
</a>
</h6>"
`;
